src/Microsoft.AspNetCore.SignalR.Protocols.MsgPack/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -38,7 +38,24 @@ public MessagePackHubProtocol()
 
         public MessagePackHubProtocol(IOptions<MessagePackHubProtocolOptions> options)
         {
-            SerializationContext = options.Value.SerializationContext;
+            var msgPackOptions = options.Value;
+            if (msgPackOptions.FormatterResolvers.Count != SignalRResolver.Resolvers.Count)
+            {
+                _resolver = new CombinedResolvers(msgPackOptions.FormatterResolvers);
+                return;
```

Comment:
`return` in a constructor? I don't think I've ever seen that. I don't see why it wouldn't work, but I'm not a fan for some arbitrary reason I can't really articulate... Can we put this in a helper that creates the resolver instead?